doc/src/manual/types.md

```diff
@@ -1127,6 +1058,146 @@ must always be specified in full, this is not especially helpful, but in Julia,
 to write just `Vector` for the abstract type including all one-dimensional dense arrays of any
 element type.
 
+## [Singleton types](@id man-singleton-types)
+
+Immutable composite types with no fields are called *singletons*. Formally, if
+
+1. `T` is an immutable composite type (ie defined with `struct`),
+1. `a isa T && b isa T` implies `a === b`,
+
+then `T` is a singleton type.[^2] [`Base.issingletontype`](@ref) can be used to check if a
```

Comment:
Is there an official convention on whether the footnote sign should go before or after the period? I saw both in the manual.

Reply:
The [Chicago Manual of Style](https://www.chicagomanualofstyle.org/qanda/data/faq/topics/Punctuation/faq0020.html) suggests that it goes after the period.

Reply:
Incidentally, since these questions are in general important for high-quality documentation, I started some discussion about referring a style guide in #36723.